In [34]:
import os
os.chdir(os.getcwd())

## Import Module

In [35]:
import pandas as pd
from scipy.stats import chi2_contingency
from scipy.stats import chi2
import numpy as np
import math

## Import Data

In [36]:
df_money = pd.read_excel('Expense-Records_6210501001.xlsx',sheet_name="Sheet2")  # ตัวอย่าง
df_money

date                                               Item  Output
0  2022-01-04                                  เงินสำหรับใช้จ่าย       0
1  2022-01-05                                                  -       0
2  2022-01-06                                                  -       0
3  2022-01-07                                  นมเปรี้ยวดัชมิลล์      35
4  2022-01-08                                            ยาคูลล์      45
5  2022-01-09                                                  -       0
6  2022-01-10                                                  -       0
7  2022-01-11                                                  -       0
8  2022-01-12                                                  -       0
9  2022-01-13                                                  -       0
10 2022-01-14                                  นมเปรี้ยวดัชมิลล์      35
11 2022-01-15                                            ยาคูลล์      45
12 2022-01-16                                                  -       0
13 2022-01-17                                                  -       0
14 2022-01-18                                                  -       0
15 2022-01-19                                                  -       0
16 2022-01-20                                                  -       0
17 2022-01-21                                  นมเปรี้ยวดัชมิลล์      35
18 2022-01-22                                            ยาคูลล์      45
19 2022-01-23                                                  -       0
20 2022-01-24                                                  -       0
21 2022-01-25                                                  -       0
22 2022-01-26                                                  -       0
23 2022-01-27                                                  -       0
24 2022-01-28                                  นมเปรี้ยวดัชมิลล์      35
25 2022-01-29                                            ยาคูลล์      45
26 2022-01-30                                                  -       0
27 2022-01-31                                                  -       0
28 2022-02-01                                                  -       0
29 2022-02-02                                                  -       0
30 2022-02-03                                                  -       0
31 2022-02-04                                  นมเปรี้ยวดัชมิลล์      35
32 2022-02-05                                            ยาคูลล์      45
33 2022-02-05  หนังสือ พัฒนา Application ด้วย React และ React...     299
34 2022-02-06                                  เงินสำหรับใช้จ่าย       0
35 2022-02-07                            ค่าเดินทางเข้ามากรุงเทพ     535
36 2022-02-07                                    ค่าอาหารกลางวัน      30
37 2022-02-08                                                  -       0
38 2022-02-09                                                  -       0
39 2022-02-10                                                  -       0
40 2022-02-11                                  นมเปรี้ยวดัชมิลล์      35
41 2022-02-12                                                  -       0
42 2022-02-13                                                  -       0
43 2022-02-14                                                  -       0
44 2022-02-15                                                  -       0
45 2022-02-16                                อุปกรณืเครื่องเขียน     285
46 2022-02-17                                                  -       0
47 2022-02-18                                  นมเปรี้ยวดัชมิลล์      35
48 2022-02-19                                            ยาคูลล์      45
49 2022-02-20                                                  -       0
50 2022-02-21                                                  -       0
51 2022-02-22                                                  -       0
52 2022-02-23                                                  -       0
53 2022-02-24                                                  -   

In [37]:
df_jan = df_money[0:28]
df_jan


date               Item  Output
0  2022-01-04  เงินสำหรับใช้จ่าย       0
1  2022-01-05                  -       0
2  2022-01-06                  -       0
3  2022-01-07  นมเปรี้ยวดัชมิลล์      35
4  2022-01-08            ยาคูลล์      45
5  2022-01-09                  -       0
6  2022-01-10                  -       0
7  2022-01-11                  -       0
8  2022-01-12                  -       0
9  2022-01-13                  -       0
10 2022-01-14  นมเปรี้ยวดัชมิลล์      35
11 2022-01-15            ยาคูลล์      45
12 2022-01-16                  -       0
13 2022-01-17                  -       0
14 2022-01-18                  -       0
15 2022-01-19                  -       0
16 2022-01-20                  -       0
17 2022-01-21  นมเปรี้ยวดัชมิลล์      35
18 2022-01-22            ยาคูลล์      45
19 2022-01-23                  -       0
20 2022-01-24                  -       0
21 2022-01-25                  -       0
22 2022-01-26                  -       0
23 2022-01-27                  -       0
24 2022-01-28  นมเปรี้ยวดัชมิลล์      35
25 2022-01-29            ยาคูลล์      45
26 2022-01-30                  -       0
27 2022-01-31                  -       0

In [38]:
df_jan = df_jan.groupby(by=["date"]).sum()
df_jan


Output
date              
2022-01-04       0
2022-01-05       0
2022-01-06       0
2022-01-07      35
2022-01-08      45
2022-01-09       0
2022-01-10       0
2022-01-11       0
2022-01-12       0
2022-01-13       0
2022-01-14      35
2022-01-15      45
2022-01-16       0
2022-01-17       0
2022-01-18       0
2022-01-19       0
2022-01-20       0
2022-01-21      35
2022-01-22      45
2022-01-23       0
2022-01-24       0
2022-01-25       0
2022-01-26       0
2022-01-27       0
2022-01-28      35
2022-01-29      45
2022-01-30       0
2022-01-31       0

In [39]:
df_feb = df_money[28:58]
df_feb


date                                               Item  Output
28 2022-02-01                                                  -       0
29 2022-02-02                                                  -       0
30 2022-02-03                                                  -       0
31 2022-02-04                                  นมเปรี้ยวดัชมิลล์      35
32 2022-02-05                                            ยาคูลล์      45
33 2022-02-05  หนังสือ พัฒนา Application ด้วย React และ React...     299
34 2022-02-06                                  เงินสำหรับใช้จ่าย       0
35 2022-02-07                            ค่าเดินทางเข้ามากรุงเทพ     535
36 2022-02-07                                    ค่าอาหารกลางวัน      30
37 2022-02-08                                                  -       0
38 2022-02-09                                                  -       0
39 2022-02-10                                                  -       0
40 2022-02-11                                  นมเปรี้ยวดัชมิลล์      35
41 2022-02-12                                                  -       0
42 2022-02-13                                                  -       0
43 2022-02-14                                                  -       0
44 2022-02-15                                                  -       0
45 2022-02-16                                อุปกรณืเครื่องเขียน     285
46 2022-02-17                                                  -       0
47 2022-02-18                                  นมเปรี้ยวดัชมิลล์      35
48 2022-02-19                                            ยาคูลล์      45
49 2022-02-20                                                  -       0
50 2022-02-21                                                  -       0
51 2022-02-22                                                  -       0
52 2022-02-23                                                  -       0
53 2022-02-24                                                  -       0
54 2022-02-25                                  นมเปรี้ยวดัชมิลล์      70
55 2022-02-26                                            ยาคูลล์      45
56 2022-02-27                                                  -       0
57 2022-02-28                                                  -       0

In [40]:
df_feb = df_feb.groupby(by=["date"]).sum()
df_feb


Output
date              
2022-02-01       0
2022-02-02       0
2022-02-03       0
2022-02-04      35
2022-02-05     344
2022-02-06       0
2022-02-07     565
2022-02-08       0
2022-02-09       0
2022-02-10       0
2022-02-11      35
2022-02-12       0
2022-02-13       0
2022-02-14       0
2022-02-15       0
2022-02-16     285
2022-02-17       0
2022-02-18      35
2022-02-19      45
2022-02-20       0
2022-02-21       0
2022-02-22       0
2022-02-23       0
2022-02-24       0
2022-02-25      70
2022-02-26      45
2022-02-27       0
2022-02-28       0

## January

In [41]:
t_day_jan = 28

In [42]:
mean_jan = df_jan['Output'].mean() #mean
mean_jan

11.428571428571429

In [43]:
sd_jan = df_jan['Output'].std()  # std
sd_jan

18.601928556785342

## February

In [44]:
t_day_feb = 28

In [45]:
mean_feb = df_feb['Output'].mean()  # mean
mean_feb

52.107142857142854

In [46]:
sd_feb = df_feb['Output'].std()  # std
sd_feb


129.86983206435306

## Hypothesis Test

สมมติว่าชุดข้อมุลดังกล่าวเป็น Normal distribution
พิสูจน์ว่าค่าใช้จ่ายเฉลี่ยในเดือนมกราคมกับเดือนกุมภาพันธ์เท่ากันหรือไม่ โดย 
- H0: ค่าใช้จ่ายเฉลี่ยในเดือนมกราคมกับเดือนกุมภาพันธ์เท่ากัน
- Ha: ค่าใช้จ่ายเฉลี่ยในเดือนมกราคมกับเดือนกุมภาพันธ์ไม่เท่ากัน

จาก Test statistic: t = (x - y − Δ0) / sqrt(s1^2/m + s2^2/n)

In [47]:
t_s = (mean_jan - mean_feb - 0) / math.sqrt((sd_jan**2)/len(df_jan)+(sd_feb**2)/len(df_feb))
t_s


-1.6406897053962113

degree of freedom


In [48]:
v = (((sd_jan**2)/len(df_jan))+((sd_feb**2)/len(df_feb)))**2 / ((((sd_jan**2 / len(df_jan))**2)/(len(df_jan)-1))+(((sd_feb**2/len(df_feb))**2)/(len(df_feb)-1)))
v

28.10741494871516

In [49]:
len(df_feb)


28

In [50]:
from scipy.stats import t
t.ppf(0.01/2,v)

-2.762498828032715

In [51]:
t.cdf(t_s,v)

0.05600338674476933

เนื่องจาก p-value = 0.056 > alpha = 0.01 ----> not reject null

ดังนั้นสรุปได้ว่า ค่าใช้จ่ายเฉลี่ยในเดือนมกราคมกับเดือนกุมภาพันธ์เท่ากัน